nmostesting/GenericTest.py

```diff
@@ -500,6 +609,91 @@ def do_test_authorization(self, api_name, test_name, error_code=401, error_type=
+            warning = None
+            headers = {"Authorization": "Bearer {}".format(token)}
+            valid, response = self.do_request("GET", url, headers=headers)
+
+            if not valid:
+                fail = response
+            elif response.status_code != 401 and response.status_code != 503:
+                fail = "Incorrect response code, expected 401 or 503. Received {}".format(response.status_code)
+                if "WWW-Authenticate" not in response.headers:
+                    fail = "Authorization error responses must include a 'WWW-Authenticate' header"
+                elif not response.headers["WWW-Authenticate"].startswith("Bearer "):
+                    fail = "'WWW-Authenticate' response header must begin 'Bearer'"
+                else:
+                    error_code = response.status_code
+                    valid, message = self.check_error_response("GET", response, error_code)
+                    if not valid:
+                        fail = message
+
+            # if node responds with 401, node should attempt to obtain the missing public key via the the token
+            # iss claim, as specified in RFC 8414 section 3.
+            # https://specs.amwa.tv/is-10/releases/v1.0.0/docs/4.5._Behaviour_-_Resource_Servers.html#public-keys
+            elif response.status_code == 401:
+                warning = "should attempt to obtain the missing public key via the the token iss claim"
+
+                error_type = "invalid_token"
+                # Remove 'Bearer ' and tokenise
+                # https://tools.ietf.org/html/rfc6750#section-3
+                error_header_ok = False
+                auth_params = response.headers["WWW-Authenticate"][7:].split(",")
+                for param in auth_params:
+                    param_parts = param.split("=")
+                    if param_parts[0] == "error":
+                        if param_parts[1] == error_type:
+                            error_header_ok = True
+                if not error_header_ok:
+                    warning = "'WWW-Authenticate' response header should contain 'error={}'".format(error_type)
+
+            # if node responds with 503 retry GET the Retry-After value
+            # https://specs.amwa.tv/is-10/releases/v1.0.0/docs/4.5._Behaviour_-_Resource_Servers.html#public-keys
+            elif response.status_code == 503:
+                # get retry-after from response
+                retry_after = response.headers.get("Retry-After")
+                if retry_after is None:
+                    warning = "'Retry-After' response should include a 'Retry-After' header"
+                else:
+                    delay = int(retry_after)
+                    time.sleep(delay)
+
+                    # do retry GET
+                    valid, response = self.do_request("GET", url, headers=headers)
+                    if not valid:
+                        fail = response
+                    elif response.status_code != 200:
```

Comment:
Is the Node not allowed to respond with `503` more than once?

Reply:
Yes, the Node can respond with `503` more than once, but the previous `503` response has already specified the `Retry-After` value, which we have already waited for that time before retrying. So I don't think the server should respond with another `503` again.

Reply:
OK, let's just tweak fail message...